src/analyses/uncaught_exceptions_analysis.cpp

```diff
@@ -24,8 +25,9 @@ irep_idt uncaught_exceptions_domaint::get_exception_type(const typet &type)
 {
   PRECONDITION(type.id()==ID_pointer);
 
-  if(type.subtype().id() == ID_struct_tag)
-    return to_struct_tag_type(type.subtype()).get_identifier();
+  if(to_pointer_type(type).base_type().id() == ID_struct_tag)
+    return to_struct_tag_type(to_pointer_type(type).base_type())
+      .get_identifier();
```

Comment:
Could we please change the interface here to just take a `const pointer_typet &`?

Reply:
done